### PR TITLE
Pie-chart

### PR DESCRIPTION
**Pull request by ChristianBeilschmidt**
- [x] I added an entry to [`CHANGELOG.md`](CHANGELOG.md) if knowledge of this change could be valuable to users.

---

Here is a brief summary of what I did:

I created a pie chart operator that uses a vector operator as input and computes pies for the distinct count of a column.

### CODE REVIEW

File: datatypes/src/plots/pie_chart.rs [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+        ensure!(
+            !legend_label.is_empty(),
+            error::Plot {
+                details: "Legend label in the pie chart must not be empty".to_string()
+            }
+        );
+        ensure!(
+            pies.values().all(|&v| v > 0.0),
+            error::Plot {
+                details: "All pies of the pie chart must have a positive value".to_string()
```
Comment (jdroenner) [inline datatypes/src/plots/pie_chart.rs:29]: Should that be "slices"? Like in slice of pizza?

File: datatypes/src/plots/pie_chart.rs [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+use serde::ser::SerializeMap;
+use serde::{Deserialize, Serialize};
+use snafu::ensure;
+use std::collections::BTreeMap;
+
+/// A pie chart
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PieChart {
+    pies: BTreeMap<String, f64>,
```
Comment (jdroenner) [inline datatypes/src/plots/pie_chart.rs:13]: maybe "slices"?

**Review by jdroenner — CHANGES_REQUESTED** [review]

File: operators/src/plot/pie_chart.rs [hunk 3]
```diff
@@ -0,0 +1,817 @@
...
+use geoengine_datatypes::primitives::{FeatureDataRef, Measurement, VectorQueryRectangle};
+use serde::{Deserialize, Serialize};
+use snafu::Snafu;
+use std::collections::HashMap;
+use tracing::{span, Level};
+
+pub const PIE_CHART_OPERATOR_NAME: &str = "PieChart";
+
+/// If the number of pies in the result is greater than this, the operator will fail.
+pub const MAX_NUMBER_OF_PIES: usize = 32;
```
[the remark this answers is not in the document]
Comment (ChristianBeilschmidt) [reply operators/src/plot/pie_chart.rs]: [0a40dc3](https://github.com/geo-engine/geoengine/pull/694/commits/0a40dc39413b8c2146aa7e02fb3815c86a28d9ec)

File: datatypes/src/plots/pie_chart.rs (hunk 2 above)
Comment (ChristianBeilschmidt) [reply datatypes/src/plots/pie_chart.rs]: [0a40dc3](https://github.com/geo-engine/geoengine/pull/694/commits/0a40dc39413b8c2146aa7e02fb3815c86a28d9ec)

**Review by jdroenner — APPROVED** [review]

**Comment by jdroenner** [comment]
bors merge

**Pull request merged**